diff --git a/pkg/chunk/cached_store.go b/pkg/chunk/cached_store.go
index ebca2031..9eaa8ea2 100644
--- a/pkg/chunk/cached_store.go
+++ b/pkg/chunk/cached_store.go
@@ -150,7 +150,8 @@ func (s *rSlice) ReadAt(ctx context.Context, page *Page, off int) (n int, err er
 	s.store.cacheMiss.Add(1)
 	s.store.cacheMissBytes.Add(float64(len(p)))
 
-	if s.store.seekable && boff > 0 && len(p) <= blockSize/4 {
+	if s.store.seekable &&
+		(!s.store.conf.CacheEnabled() || (boff > 0 && len(p) <= blockSize/4)) {
 		if s.store.downLimit != nil {
 			s.store.downLimit.Wait(int64(len(p)))
 		}
